app/mailers/promo_mailer.rb

```diff
@@ -0,0 +1,19 @@
+class PromoMailer < ApplicationMailer
+  include PublishersHelper
+
+  def activate_promo_2018q1(publisher)
+    @publisher = publisher
+    @private_promo_2018q1_auth_url = generate_private_promo_2018q1_auth_url(publisher)
+
+    mail(
+      to: @publisher.email,
+      subject: default_i18n_subject
+    )
+  end
+
+  # Maybe move this to publisher helper?
+  def generate_private_promo_2018q1_auth_url(publisher)
```

Comment:
agree, generally mailers should be pretty devoid of state changes
I think we should generate the promo activate tokens in a rake task.

to get the url, you could create a PromoHelper and make a `publisher_promo_auth_url` which takes a publisher and promo id